Pull Request for Full Review

PlagueHO opened the pull request:
PR for performing a full module review.

See: https://github.com/PowerShell/DscResources/issues/74

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/plagueho/iscsidsc/16)

<!-- Reviewable:end -->

kwirkykat commented:
Reviewed 2 of 22 files at r1.
Review status: 2 of 22 files reviewed at latest revision, 27 unresolved discussions.

---

*[appveyor.yml, line 4 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-D4mXxG-EAssR9vi1:-KP-D4mXxG-EAssR9vi2:b-utqs7d) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/appveyor.yml#L4)):*
> ```YAML
> #      environment configuration  #
> #---------------------------------#
> version: 1.1.{build}.0
> ```

Version should be 1.3.{build}.0

---

*[appveyor.yml, line 10 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-CNZXrCorqe7xrYqW:-KP-CNZXrCorqe7xrYqX:b-668h72) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/appveyor.yml#L10)):*
> ```YAML
>         Push-Location
>         cd DscResource.Tests
>         Import-Module .\TestHelper.psm1 -force
> ```

Remove location change by combining it with module import:
```powershell
Import-Module .\DscResource.Tests\TestHelper.psm1 -Force
```

---

*[appveyor.yml, line 44 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-D1LiIzoUiRmIbCkF:-KP-D1LiIzoUiRmIbCkG:bw1aahp) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/appveyor.yml#L44)):*
> ```YAML
>       $stagingDirectory = (Resolve-Path ..).Path
>       $manifest = Join-Path $pwd "iSCSIDsc.psd1"
>       (Get-Content $manifest -Raw).Replace("1.0.0.0", $env:APPVEYOR_BUILD_VERSION) | Out-File $manifest
> ```

Version should be 1.3.0.0

---

*[iSCSIDsc.psd1, line 7 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-CpkzcMXJSyJfuO9f:-KP-Cpl-y0NdwQ8dDMTO:b-xkrx2g) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/iSCSIDsc.psd1#L7)):*
> ```PowerShell
> 
> # Version number of this module.
> ModuleVersion = '1.2.1.0'
> ```

ModuleVersion = '1.3.0.0'

---

*[iSCSIDsc.psd1, line 19 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-CtDlcYiqot57brVX:-KP-CtDlcYiqot57brVY:bt53p86) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/iSCSIDsc.psd1#L19)):*
> ```PowerShell
> 
> # Copyright statement for this module
> Copyright = '(c) 2015 Microsoft Corporation. All rights reserved.'
> ```

Update to 2016

---

*[README.md, line 6 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-DPnoNOmQVdRLFzV8:-KP-DPnoNOmQVdRLFzV9:bkm9c42) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L6)):*
> ```Markdown
> 
> The **iSCSIDsc** module contains DSC resources for configuring Windows iSCSI Targets and Initiators.
> ```

Please add snippet about the MS open source code of conduct:
```
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
```

---

*[README.md, line 8 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-EodTh5pHjuCqW68P:-KP-EodTh5pHjuCqW68Q:ba0j8et) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L8)):*
> ```Markdown
> 
> ## Requirements
> - The iSCSI Target resources can only be used on Windows Server 2012 and above. They can not be used Windows Desktop operating systems.
> ```

... They can not be used **on** Windows Desktop operating systems. ...

---

*[README.md, line 14 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-FLT0wWDFZPx_s1VK:-KP-FLT14hbUlbC0ze5V:bxi0plb) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L14)):*
> ```Markdown
> ## Installation
> ```powershell
> Install-Module -Name iSCSIDsc -MinimumVersion 1.2.1.0
> ```

Can you take out the MinimumVersion parameter on this call? So that we don't have to update this line for each release

---

*[README.md, line 39 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-IQNny8CaA74iBBzq:-KP-IQNny8CaA74iBBzr:b-3p66hj) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L39)):*
> ```Markdown
> 
> #### Parameters
> - **Ensure**: Ensures that Virtual Disk is either Absent or Present. Required.
> ```

Add that the default value is Present.
This parameter is also Optional, not required

---

*[README.md, line 41 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-Igg7GhotkWNE1LIg:-KP-Igg7GhotkWNE1LIh:b-82rsum) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L41)):*
> Required.
This is not a mandatory parameter in Set/Test-TargetResource

---

*[README.md, line 42 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-IxPJg59u5atjRhRk:-KP-IxPJg59u5atjRhRl:bm0fyyg) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L42)):*
> the default size if 2 MB
is

---

*[README.md, line 43 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-IYE0TZ9z4kOid5GY:-KP-IYE0TZ9z4kOid5GZ:b-a40plc) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L43)):*
> Dynamics
Dynamic

---

*[README.md, line 53 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-JOtl_iSFkydpgKgi:-KP-JOtl_iSFkydpgKgj:bg6jc95) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L53)):*
> ```Markdown
> 
> #### Parameters
> - **Ensure**: Ensures that Server Target is either Absent or Present. Required.
> ```

Add that the default value is Present
This parameter is also Optional, not required

---

*[README.md, line 61 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-N3Z9sC7UGm0Bl7fU:-KP-N3ZAV7vOB1ce8hh2:bjaqupp) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L61)):*
> iSCSI Target Portals
iSCSI Target Portal

> connect to an iSCSI Targets on them
connect it to an iSCSI Target

---

*[DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1, line 442 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-bOXd-_vy5WuVpPpP:-KP-bOXeZmI5nL3p7Geq:b-6gqkmw) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1#L442)):*
> ```PowerShell
> # Helper Functions
> 
> Function Get-ServerTarget
> ```

Please add a help comment above this function.

---

*[DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1, line 448 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-bV-18vMAwJqRSMtL:-KP-bV-18vMAwJqRSMtM:b-puw67k) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1#L448)):*
> ```PowerShell
>         [parameter(Mandatory = $true)]
>         [System.String]
>         $TargetName
> ```

This parameter should have the ValidateNotNullOrEmpty attribute

---

*[DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1, line 453 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-biao2xT_u2xn7F_g:-KP-biao2xT_u2xn7F_h:b-bl3gav) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1#L453)):*
> ```PowerShell
>     {
>         $ServerTarget = Get-iSCSIServerTarget `
>             -ComputerName LOCALHOST `
> ```

ComputerName should not be hardcoded. Perhaps you should add ComputerName as a parameter to this function with a default value of 'localhost'  or perhaps should be $env:computerName

---

*[DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1, line 397 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-cpp5v7SkgLVyTdib:-KP-cpp5v7SkgLVyTdic:b-6gqkmw) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1#L397)):*
> ```PowerShell
> # Helper Functions
> 
> Function Get-VirtualDisk
> ```

Please add a help comment above this function.

---

*[DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1, line 403 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-cvmPHHwXfydFxqOj:-KP-cvmPHHwXfydFxqOk:b-puw67k) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1#L403)):*
> ```PowerShell
>         [parameter(Mandatory = $true)]
>         [System.String]
>         $Path
> ```

This parameter should have the ValidateNotNullOrEmpty attribute

---

*[DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1, line 410 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-cc6LjhrHYEWt3II7:-KP-cc6LjhrHYEWt3II8:ba1oey5) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1#L410)):*
> ```PowerShell
>         # it speeds cmdlet up significantly
>         $VirtualDisk = Get-iSCSIVirtualDisk `
>             -ComputerName LOCALHOST `
> ```

ComputerName should not be hardcoded. Perhaps you should add ComputerName as a parameter to this function with a default value of 'localhost' or perhaps should be $env:computerName

---

*[Examples/Sample_iSCSIInitiator.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-dX_cE6HgEVSQSuQs:-KP-dX_cE6HgEVSQSuQt:b-2eihn7) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Examples/Sample_iSCSIInitiator.ps1#L1)):*
> ```PowerShell
> configuration Sample_iSCSIInitiator
> ```

Please put a comment at the top of the example that explains what it does and what is required to run it.

---

*[Examples/Sample_iSCSIServerTarget.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-dPoFlYE-i6iAcM0L:-KP-dPoFlYE-i6iAcM0M:b-2eihn7) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Examples/Sample_iSCSIServerTarget.ps1#L1)):*
> ```PowerShell
> configuration Sample_ciSCSIServerTarget
> ```

Please put a comment at the top of the example that explains what it does and what is required to run it.

---

*[Tests/Integration/MSFT_iSCSIInitiator.config.ps1, line 3 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-dlZ1Qc_fucvj3SD1:-KP-dlZ1Qc_fucvj3SD2:b-4keqju) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIInitiator.config.ps1#L3)):*
> ```PowerShell
> $IPAddress = (Get-NetIPAddress -InterfaceIndex (Get-NetConnectionProfile -IPv4Connectivity Internet).InterfaceIndex -AddressFamily IPv4).IPAddress
> $TargetName = 'TestServerTarget'
> $Initiator = @{
> ```

Why have the extra Initiator object declared up here with the same properties as the resource instead of just filling in the properties of the resource in the configuration?

---

*[Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1, line 77 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-e8yEE4XajaH2zIJP:-KP-e8yEE4XajaH2zIJQ:b-7401vu) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1#L77)):*
> # Create a Server Target on this computer to test with
<details><summary><i alt="summary">Quoted 13 lines of code…</i></summary>
>         $VirtualDiskPath = Join-Path -Path $ENV:Temp -ChildPath ([System.IO.Path]::ChangeExtension([System.IO.Path]::GetRandomFileName(),'vhdx'))
>         New-iSCSIVirtualDisk `
>             -ComputerName LOCALHOST `
>             -Path $VirtualDiskPath `
>             -SizeBytes 500MB
>         New-iSCSIServerTarget `
>             -TargetName $TargetName `
>             -InitiatorIds "Iqn:iqn.1991-05.com.microsoft:$($Initiator.InitiatorPortalAddress)" `
>             -ComputerName LOCALHOST
>         Add-IscsiVirtualDiskTargetMapping `
>             -ComputerName LOCALHOST `
>             -TargetName $TargetName `
>             -Path $VirtualDiskPath
</details>
This code should be contained in a Pester BeforeAll block

---

*[Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1, line 144 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-eHBR4tGFRrOh6WbE:-KP-eHBR4tGFRrOh6WbF:bobi9c3) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1#L144)):*
>  Disconnect-IscsiTarget `
<details><summary><i alt="summary">Quoted 15 lines of code…</i></summary>
>             -NodeAddress $Initiator.NodeAddress `
>             -Confirm:$False
>         Remove-IscsiTargetPortal `
>             -TargetPortalAddress $Initiator.TargetPortalAddress `
>             -InitiatorPortalAddress $Initiator.InitiatorPortalAddress `
>             -Confirm:$False
>         Remove-iSCSIServerTarget `
>             -ComputerName LOCALHOST `
>             -TargetName $TargetName
>         Remove-iSCSIVirtualDisk `
>             -ComputerName LOCALHOST `
>             -Path $VirtualDiskPath
>         Remove-Item `
>             -Path $VirtualDiskPath `
>             -Force
</details>
This code should be contained in a Pester AfterAll block

---

*[Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1, line 52 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-f3MykYxuPBTJYgYZ:-KP-f3Mz8wYX5QhH6smf:b3ucavy) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1#L52)):*
> $ConfigFile = Join-Path -Path $PSScriptRoot -ChildPath "$($Global:DSCResourceName).config.ps1"
>     . $ConfigFile
This code should probably go inside the Pester BeforeAll block for the Describe block below.

---

*[Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1, line 82 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-fDFBGbtGNpAZCF6j:-KP-fDFBGbtGNpAZCF6k:b-s4b6gr) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1#L82)):*
> # Clean up
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>
>         Remove-iSCSIVirtualDisk `
>             -Path $VirtualDisk.Path
>         Remove-Item `
>             -Path $VirtualDisk.Path `
>             -Force
</details>
This code should be contained in a Pester AfterAll block.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat  - thank you heaps for reviewing this. I'll get onto making the changes over the next couple of days (have run out of time tonight).

PlagueHO commented:
I've done some work on the changes but have run out of time again tonight unfortunately. So I'll continue with the updates tomorrow. Thanks again for the review.

Also, I've moved the localization data out into external files like all the other resources for consistency.

---

Review status: 1 of 25 files reviewed at latest revision, 27 unresolved discussions.

---

*[appveyor.yml, line 4 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-D4mXxG-EAssR9vi1:-KPMUfPmQrDRPtnaPOJw:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/appveyor.yml#L4)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Version should be 1.3.{build}.0
</details>
Done.

---

*[appveyor.yml, line 10 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-CNZXrCorqe7xrYqW:-KPMUo0rtFSckdRPlzRN:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/appveyor.yml#L10)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Remove location change by combining it with module import:
> ```powershell
> Import-Module .\DscResource.Tests\TestHelper.psm1 -Force
> ```
</details>
Done.

---

*[appveyor.yml, line 44 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-D1LiIzoUiRmIbCkF:-KPMUpBsGQ8Ci4G_xtXD:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/appveyor.yml#L44)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Version should be 1.3.0.0
</details>
Done.

---

*[iSCSIDsc.psd1, line 7 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-CpkzcMXJSyJfuO9f:-KPMUrEtuuojTuAs2xTL:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/iSCSIDsc.psd1#L7)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> ModuleVersion = '1.3.0.0'
</details>
Done.

---

*[iSCSIDsc.psd1, line 19 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-CtDlcYiqot57brVX:-KPMUx7Fa4McN-eXXrAf:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/iSCSIDsc.psd1#L19)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Update to 2016
</details>
Done.

---

*[README.md, line 6 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-DPnoNOmQVdRLFzV8:-KPMUyTL59zqUH4229R1:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L6)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add snippet about the MS open source code of conduct:
> ```
> This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
> For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
> ```
</details>
Done.

---

*[README.md, line 8 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-EodTh5pHjuCqW68P:-KPMVMJbgNKzB8iwLfs5:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L8)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> ... They can not be used **on** Windows Desktop operating systems. ...
</details>
Done.

---

*[README.md, line 14 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-FLT0wWDFZPx_s1VK:-KPMVP_lwQy6_sys4XXZ:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L14)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Can you take out the MinimumVersion parameter on this call? So that we don't have to update this line for each release
</details>
Done.

---

*[README.md, line 39 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-IQNny8CaA74iBBzq:-KPMV_7sb4w6Kc7S9EWY:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L39)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Add that the default value is Present.
> This parameter is also Optional, not required
</details>
Done.

---

*[README.md, line 41 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-Igg7GhotkWNE1LIg:-KPMX_XKzfhPcKiRlHjp:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L41)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > Required.
> This is not a mandatory parameter in Set/Test-TargetResource
</details>
Done.

---

*[README.md, line 42 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-IxPJg59u5atjRhRk:-KPMXke1TezPDzQgPoCh:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L42)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > the default size if 2 MB
> is
</details>
Done.

---

*[README.md, line 43 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-IYE0TZ9z4kOid5GY:-KPMXlLNzIBtfuwzkDOa:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L43)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > Dynamics
> Dynamic
</details>
Done.

---

*[README.md, line 53 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-JOtl_iSFkydpgKgi:-KPMY7fxUIkjM2I9hL7M:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L53)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Add that the default value is Present
> This parameter is also Optional, not required
</details>
Done.

---

*[README.md, line 61 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-N3Z9sC7UGm0Bl7fU:-KPMYUdn4VlUDI3rLtKO:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/README.md#L61)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > iSCSI Target Portals
> iSCSI Target Portal
> 
> > connect to an iSCSI Targets on them
> connect it to an iSCSI Target
</details>
Done.

---

*[Examples/Sample_iSCSIInitiator.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-dX_cE6HgEVSQSuQs:-KPMZ16mYDMIBBmQ_HKK:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Examples/Sample_iSCSIInitiator.ps1#L1)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please put a comment at the top of the example that explains what it does and what is required to run it.
</details>
Done.

---

*[Examples/Sample_iSCSIServerTarget.ps1, line 1 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-dPoFlYE-i6iAcM0L:-KPMZRnBsNOjy_kP_Rot:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Examples/Sample_iSCSIServerTarget.ps1#L1)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please put a comment at the top of the example that explains what it does and what is required to run it.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-:-KPM_V8GdQ1H4kDEC-UY:b-7ar1e7)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
I've added Get-Help info to all the functions now.

---

Review status: 1 of 25 files reviewed at latest revision, 27 unresolved discussions.

---

*[DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1, line 442 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-bOXd-_vy5WuVpPpP:-KPMdEJoPizriVAjp59w:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1#L442)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add a help comment above this function.
</details>
Done.

---

*[DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1, line 397 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-cpp5v7SkgLVyTdib:-KPMdOTI3ZEINXKkq0-x:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1#L397)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please add a help comment above this function.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-:-KPMg1iUsOxcXngQrUSm:b-ec20co)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Review status: 1 of 25 files reviewed at latest revision, 27 unresolved discussions.

---

*[DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1, line 448 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-bV-18vMAwJqRSMtL:-KPMgtMmXI9a-dZbVM9l:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1#L448)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This parameter should have the ValidateNotNullOrEmpty attribute
</details>
Done.

---

*[DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1, line 403 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-cvmPHHwXfydFxqOj:-KPMh7f8OfS26XsaY8Cn:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1#L403)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This parameter should have the ValidateNotNullOrEmpty attribute
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
I haven't quite finished this yet. I ran out of time this weekend. Will try and complete the PR tomorrow night after work.

---

Review status: 1 of 25 files reviewed at latest revision, 27 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-:-KPgSigtXfQBQxhTUWXP:b3smtp8)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat - I've completed the work on this as per your comments (although I've raised some queries on some of them). I've also adjusted the unit/integration tests to use the standard 1.1.0 header and also removed the Global variables.

---

Review status: 1 of 25 files reviewed at latest revision, 27 unresolved discussions.

---

*[DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1, line 453 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-biao2xT_u2xn7F_g:-KPkqh9x2HCKmLrXVeqw:b4o4zwh) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIServerTarget/MSFT_iSCSIServerTarget.psm1#L453)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> ComputerName should not be hardcoded. Perhaps you should add ComputerName as a parameter to this function with a default value of 'localhost'  or perhaps should be $env:computerName
</details>
Normally I'm 100% in agreement here, but the problem is that there seems to some irritating behavior with the iSCSI cmdlets:
1. If you don't specify a ComputerName (it's an optional parameter) then there is a 20-30 second delay for the cmdlet to complete.
2. If you specify a ComputerName with the FQDN or the name of the Computer (e.g. $ENV:ComputerName) then there is a 20-30 second delay for the cmdlet to complete as well.

I wrote a blog post detailing the problem here: https://dscottraynsford.wordpress.com/2016/01/02/speed-up-iscsi-powershell-cmdlets/

I didn't expose this as a parameter because people might try to use it - which defeats the purpose of DSC to some degree (you don't want DSC resources to configure other nodes).

I'd actually prefer to not have set the ComputerName at all, but a 20-30 second delay (often many many times per config for machines with lots of iSCSI disks) would be really bad.

I'm happy to define a $ComputerName = 'LocalHost' variable and use that, but I don't think it should be exposed to be configured. It sort of felt pointless to make it a variable though because I'm hard coding it completely intentionally: because I don't want it to be changed - ever :smile:

Will take your instruction though of course!

---

*[DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1, line 410 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-cc6LjhrHYEWt3II7:-KPkt2yFr98xFgQqwUlo:bu6xv8r) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/DSCResources/MSFT_iSCSIVirtualDisk/MSFT_iSCSIVirtualDisk.psm1#L410)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> ComputerName should not be hardcoded. Perhaps you should add ComputerName as a parameter to this function with a default value of 'localhost' or perhaps should be $env:computerName
</details>
See previous comments

---

*[Tests/Integration/MSFT_iSCSIInitiator.config.ps1, line 3 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-dlZ1Qc_fucvj3SD1:-KPktRaR0CMH55Tps2_R:b-hx5b15) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIInitiator.config.ps1#L3)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Why have the extra Initiator object declared up here with the same properties as the resource instead of just filling in the properties of the resource in the configuration?
</details>
It is so that the Object is available to the Integration test itself. E.g. $Initator can be accessed and used in MSFT_iSCSIInitiator.Integration.Tests to confirm that the DSC config has correctly applied the changes. Most integration tests should be using this design pattern.

---

*[Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1, line 77 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-e8yEE4XajaH2zIJP:-KPl-NKurTZNZC2LhHz2:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1#L77)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > # Create a Server Target on this computer to test with
> >         $VirtualDiskPath = Join-Path -Path $ENV:Temp -ChildPath ([System.IO.Path]::ChangeExtension([System.IO.Path]::GetRandomFileName(),'vhdx'))
> >         New-iSCSIVirtualDisk `
> >             -ComputerName LOCALHOST `
> >             -Path $VirtualDiskPath `
> >             -SizeBytes 500MB
> >         New-iSCSIServerTarget `
> >             -TargetName $TargetName `
> >             -InitiatorIds "Iqn:iqn.1991-05.com.microsoft:$($Initiator.InitiatorPortalAddress)" `
> >             -ComputerName LOCALHOST
> >         Add-IscsiVirtualDiskTargetMapping `
> >             -ComputerName LOCALHOST `
> >             -TargetName $TargetName `
> >             -Path $VirtualDiskPath
> This code should be contained in a Pester BeforeAll block
</details>
Done.

---

*[Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1, line 144 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-eHBR4tGFRrOh6WbE:-KPl-OlLWazOrzuDGrde:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIInitiator.Integration.Tests.ps1#L144)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> >  Disconnect-IscsiTarget `
> >             -NodeAddress $Initiator.NodeAddress `
> >             -Confirm:$False
> >         Remove-IscsiTargetPortal `
> >             -TargetPortalAddress $Initiator.TargetPortalAddress `
> >             -InitiatorPortalAddress $Initiator.InitiatorPortalAddress `
> >             -Confirm:$False
> >         Remove-iSCSIServerTarget `
> >             -ComputerName LOCALHOST `
> >             -TargetName $TargetName
> >         Remove-iSCSIVirtualDisk `
> >             -ComputerName LOCALHOST `
> >             -Path $VirtualDiskPath
> >         Remove-Item `
> >             -Path $VirtualDiskPath `
> >             -Force
> This code should be contained in a Pester AfterAll block
</details>
Done.

---

*[Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1, line 52 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-f3MykYxuPBTJYgYZ:-KPl1q9KXnHLTMhQHazd:b-f1yu09) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1#L52)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > $ConfigFile = Join-Path -Path $PSScriptRoot -ChildPath "$($Global:DSCResourceName).config.ps1"
> >     . $ConfigFile
> This code should probably go inside the Pester BeforeAll block for the Describe block below.
</details>
If we're going to do that I'd suggest we change the template first as I think it is probably a good idea to keep inline with the template. And if that were the case we'd probably want to go back and change other integration tests to this model as well. Almost all integration tests in all resources (where people have actually created them... *sigh*) have adopted this model - so for consistency we'd want to go back and change all those as well - which is a bunch more tech debt being added on (and TBH it feels like most repos are drowning in debt already). Sorry - it's been a long weekend of trying to reduce the tech debt so that we can actually add new features. I'll stop complaining now :smile:

I'll take your guidance on this of course.

---

*[Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1, line 82 \[r1\]](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-KP-fDFBGbtGNpAZCF6j:-KPl1bGqD6aFAdlTLUga:b-896fix) ([raw file](https://github.com/plagueho/iscsidsc/blob/aa2d01a226e420044798bc5d5287458408589219/Tests/Integration/MSFT_iSCSIVirtualDisk.Integration.Tests.ps1#L82)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > # Clean up
> >         Remove-iSCSIVirtualDisk `
> >             -Path $VirtualDisk.Path
> >         Remove-Item `
> >             -Path $VirtualDisk.Path `
> >             -Force
> This code should be contained in a Pester AfterAll block.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-:-KPl2MhrXUZiY4SA1u1L:b-2wqvlb)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Cool! Thanks @travisez13 

---

Review status: 4 of 25 files reviewed at latest revision, 27 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-:-KPqPED_uYsVIOVRkrIh:b-rfx7oq)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
@kwirkykat , @travisez13  - are you waiting on anything from me on this one? I'm trying really hard to clear my backlog out so I can start addressing some issues in PSDesiredStateConfiguration and xNetworking resources (plus more tech debt reduction) - working on too many open issues at once and I get confused :smile:

---

Review status: 4 of 25 files reviewed at latest revision, 27 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plagueho/iscsidsc/16#-:-KRXrEhd3r6v-ILEW1XT:brvf9ag)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@PlagueHO Nope not waiting on you. Sorry, we just have way too much work on our plates right now.
@mbreakey3 and I have to focus on xPSDSC up through the next release (Sept 21) per some promises our management made.

PlagueHO commented:
@kwirkykat : I've been there :smile: feel your pain. Don't worry about reviewing my new resources. I just wanted to make sure you weren't waiting on me to do something. Just pick then up again when the pressure is off.

Is there anything I can focus on to help with the load you lot are under? I'm currently working hard on cleaning up xstorage and adding new features to that but if there are any other specific items I could help please let me know.

PlagueHO commented:
Hi @kwirkykat - I'm a bit stuck with what to do with these repos at the moment.

I'd like to try and bring them into line with all the other work we've been doing in the DSC Resource kit. However several of my repos are stuck in the state of "limbo" and have been for a year or more :grin:

Are you able to give me an idea of what I should do with this? I really don't want to revert back to the "cISCSC" format, because as far as I'm concerned this module is HQRM (I really hate the "x" and "c" nomenclature).

I'll raise an issue in the DSCResource Kit so that we can discuss at the next community call.

kwirkykat commented:
@PlagueHO I honestly am not sure what to do with these repos anymore either. The internal conversation has been up in the air for a while now.
@zjalexander

PlagueHO commented:
Hi @kwirkykat - absolutely no problem at all :grin:

Would you guys think it would be worth trying this module (or my WSMandDsc - I've had many requests for new features on that one) as a guinea pig for the "community managed" but linked to by the resource kit type model?

I'm happy to give this a try if you guys are?
